### PR TITLE
MODPUBSUB-215: Replace Future<Boolean> by Future<Void> in ConsumerService

### PR DESCRIPTION
**Pull request by yusuf-murodov-epam**
## Purpose
[MODPUBSUB-215](https://issues.folio.org/browse/MODPUBSUB-215)
_Replace Future<Boolean> by Future<Void> in ConsumerService_

### CODE REVIEW

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java [hunk 1]
```diff
@@ -69,40 +69,28 @@ public KafkaConsumerServiceImpl(@Autowired Vertx vertx,
...
-          result.complete(true);
-        } else {
-          result.fail(ar.cause());
-        }
-      });
-    return result.future();
+        return KafkaConsumer.<String, String>create(vertx, consumerProps)
+          .handler(getEventReceivedHandler(params))
+          .subscribe(topic)
+          .onSuccess(x -> {
```
Comment (alexanderkurash) [inline mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java:85]: What is this x? Can we give it a more descriptive name?

File: mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
    *
    * @param eventTypes list of event types that specified module is subscribing to receive
    * @param params     Okapi connection params
-   * @return future with true if succeeded
+   * @return a future completed with a result
```
Comment (alexanderkurash) [inline mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java:18]: Should we make this description more direct, like in https://github.com/folio-org/mod-pubsub/pull/200/files this PR?

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java [hunk 3]
```diff
@@ -69,40 +69,28 @@ public KafkaConsumerServiceImpl(@Autowired Vertx vertx,
...
+            cache.addSubscription(topic);
+            LOGGER.info(format("Subscribed to topic {%s}", topic));
+          })
+          .onFailure(e -> {
+            LOGGER.error(format("Could not subscribe to some of the topic {%s}", topic), e);
+          });
+      }).collect(Collectors.toList());
```
Comment (alexanderkurash) [inline mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java:92]: Should this .collect be aligned with the .map and .filter on lines 79 and 80?

Comment (yusuf-murodov-epam) [reply mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java]: fixed

File: mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java (hunk 2 above)
Comment (yusuf-murodov-epam) [reply mod-pubsub-server/src/main/java/org/folio/services/ConsumerService.java]: fixed

File: mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java (hunk 1 above)
Comment (yusuf-murodov-epam) [reply mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java]: changed to 'result'

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**